### PR TITLE
[#120455] Privileged users should be able to edit, but not create split accounts

### PR DESCRIPTION
**Pull request by jhanggi**
And refactor how we extend Abilitiy.

### CODE REVIEW

File: vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb [hunk 1]
```diff
@@ -66,8 +39,61 @@
...
+  describe "as an admin" do
+    let(:user) { FactoryGirl.create(:user, :administrator) }
+
+    include_examples "allows editing"
+    include_examples "allows suspending"
+
+    describe "default new" do
+      before { get :new, facility_id: facility.url_name, owner_user_id: user.id }
+
+      it "sees split account option" do
```
Comment (drench) [inline vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb]: `:aggregate_failures`?

File: vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb [hunk 2]
```diff
@@ -66,8 +39,61 @@
...
+        expect(response.code).to eq("200")
+        expect(response.body).to include("Chart String")
+        expect(response.body).to include("Split Account")
```
Comment (drench) [inline vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb]: This feels like a candidate for `.and` chaining.

File: lib/ability_extension_manager.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class AbilityExtensionManager
```
Comment (benlinton) [inline lib/ability_extension_manager.rb]: Can we add a short comment at the top?  About how the extensions build on each other and order matters?

File: vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb (hunk 2 above)
Comment (jhanggi) [inline vendor/engines/split_accounts/spec/controllers/facility_accounts_controller_spec.rb]: I played with it, and it got super awkward for the examples with `not_to`, and I'd rather be a little verbose but remain consistent within the file. Using `:aggregate_failures` should accomplish the same upside, though, so I went ahead and added that.

**Comment by jhanggi** [comment]
Updated.

File: lib/ability_extension_manager.rb [hunk 4]
```diff
@@ -1,3 +1,17 @@
+# This class is used by the Ability class in order for you to customize the
+# ability behavior inside of an engines. Extensions should take the ability as
```
Comment (benlinton) [inline lib/ability_extension_manager.rb]: Minor typo `an engines`

**Comment by benlinton** [comment]
:+1:

**Comment by drench** [comment]
:+1:

**Pull request merged**